### PR TITLE
Color the icon background with the error color or lacking production

### LINKED ISSUE #87: Lacking Desired Products are colored as overproduction
When the specified amount of the desired product is not delivered, it is colored as purple instead of red.
Here's the minimal example that demonstrates the issue. You can do it on other :
1. Open YAFC of version 0.6.2 or higher.
2. Make a new sheet.
3. Specify that you want `-1` or `1` Copper Cables.
4. Observe that the cables are colored purple and not red.

### PR DESCRIPTION
**Pull request by veger**
I found `linkFlow` which seems to show to surplus/lack of flow within the link. By matching this to the (desired) `amount` we can detect an error (lack) of flow.

Some tests showed it working, but I am not 100% sure this is what `linkFlow` should be used for. So some additional testing for side effects wouldn't hurt...

Fixes #87

### CODE REVIEW

**Review by have-fun-was-taken — COMMENTED** [review]
The negative amount of a desired product still makes the background purple.

grafik

Steps to reproduce:
1. Make a new sheet.
2. Enter any item with a negative amount in the Desired Products.

**Comment by veger** [comment]
Oh... I forgot to test negative values :disappointed:

**Comment by have-fun-was-taken** [comment]
Alright, the "-1" case seems to be fixed, but I didn't test the usual cases much. Feel free to check if it works as expected.

**Comment by have-fun-was-taken** [comment]
@veger On a side note, if you're okay with this, I vote for the `<summary>` entries to be done as usual, in three lines, and not on one line. I value the vertical space, but it is mentally taxing to skip over the start of the string to get to the needed information.

File: YAFC/Workspace/ProductionTable/ProductionTableView.cs [hunk 1]
```diff
@@ -880,6 +880,10 @@ private enum ProductDropdownType {
             }
         }
 
+        private static bool CheckIfOverproducing(ProductionLink link) {
+            return link.algorithm == LinkAlgorithm.AllowOverProduction && (link.flags & ProductionLink.Flags.LinkNotMatched) != 0;
```
Comment (veger) [inline YAFC/Workspace/ProductionTable/ProductionTableView.cs:884]: `LinkNotMatched` could also get triggered for/with `AllowOverConsumption`, so by just checking the flags it is not possible to determine over production. That is why I included the `element.linkFlow > element.amount` check for the coloring.

So the function name does not cover its implementation? Renaming to 'CheckPossibleOverproducing` might covert its implementation better?

Comment (veger) [inline YAFC/Workspace/ProductionTable/ProductionTableView.cs:884]: There is a `HasFlags()` function to make checking for certain flags (slightly) nicer. See https://github.com/have-fun-was-taken/yafc-ce/pull/94/commits/27308c5136bfdb33b787696c0479933a2f296f8f#diff-75c10f5bf285559ced0cf39cf2285ef4f709280fb8fc34c50e1bfa57952a255aR827 for an example

**Comment by veger** [comment]
> Alright, the "-1" case seems to be fixed

Nice, I was just about to take a look at this case. I'll give it a test instead :smile: 

> On a side note, if you're okay with this, I vote for the <summary> entries to be done as usual, in three lines, and not in one line. I value the vertical space, but it is mentally taxing to skip over the start of the string to get to the needed information

Totally fine by me. Might be something for the contributors document as well, so we standardize these guidelines without the project as well?

File: YAFC/Workspace/ProductionTable/ProductionTableView.cs (hunk 1 above)
Comment (veger) [reply YAFC/Workspace/ProductionTable/ProductionTableView.cs]: After a quick test: over consuming seems to cause the solver giving issues. So maybe this flag is never triggered? :thinking:

Comment (have-fun-was-taken) [reply YAFC/Workspace/ProductionTable/ProductionTableView.cs]: Thank you for the suggestion! I've applied it in https://github.com/have-fun-was-taken/yafc-ce/pull/94/commits/e613047b3b75378854dd3abdbf1cdebbe6174c86 and https://github.com/have-fun-was-taken/yafc-ce/pull/94/commits/c89b7047f0a2ce9f585f5ba89c6d21459bc8e3c9.

Comment (have-fun-was-taken) [reply YAFC/Workspace/ProductionTable/ProductionTableView.cs]: > seems to cause the solver giving issues

Could you please elaborate? Did you mean that the magenta color is not triggered for desired products when it should?

Comment (veger) [reply YAFC/Workspace/ProductionTable/ProductionTableView.cs]: The solver cannot solve overconsumption (there is no 'allow over consumption' option on a link). So the solver will abort with the error error box on top of the sheet. (same confusion as #96  is about, I think)

**Comment by veger** [comment]
I merged the 'fixup' commits and rebased on the master branch

**Review by have-fun-was-taken — APPROVED** [review]
Checked 1, -1, and a couple of usual cases.

**Pull request merged**